packages/@react-spectrum/dialog/docs/Dialog.mdx

```diff
@@ -37,58 +46,371 @@ import {Heading, Text} from '@react-spectrum/typography';
+If your Dialog is dismissable, make sure that you do not include a ButtonGroup in the render. 
+While the Dialog will hide any elements occupying the ButtonGroup section, it will not remove them from the DOM, 
+resulting in possible tab order issues.
 
-## Value
+```tsx example
+<DialogTrigger isDismissable>
+  <ActionButton>Status</ActionButton> 
+  <Dialog>
+    <Heading>Status</Heading>
+    <Divider />
+    <Content>Printer Status: Connected</Content>
+  </Dialog>
+</DialogTrigger>
+```
 
-*If the component displays or allows a user to input a value, include this section.*
+It is important to note that the `Heading`, `Header`, `Content`, and `Footer` content elements accept any renderable 
+node, not just strings. This allows you to create Dialogs for more complex workflows, such as including a form
+for the user to fill out or adding confirmation checkboxes.
+
+```tsx example
+<DialogTrigger>
+  <ActionButton>Register</ActionButton> 
+  {(close) => (
+    <Dialog>
+      <Heading>
+        <Flex alignItems="center">
+          <Book size="S" />
+          <Text>
+            Register for newsletter
+          </Text>
+        </Flex>
+      </Heading>
+      <Header>
+        <StatusLight variant="negative">
+          Registration Status
+        </StatusLight>
+      </Header>
+      <Divider />
+      <Content>
+        <Form>
+          <TextField label="First Name" placeholder="John" autoFocus />
+          <TextField label="Last Name" placeholder="Smith" />
+          <TextField label="Street Address" placeholder="123 Any Street" />
+          <TextField label="City" placeholder="San Francisco" />
+        </Form>
+      </Content>
+      <Footer>
+        <Checkbox>
+          I want to receive updates for exclusive offers in my area.
+        </Checkbox>
+      </Footer> 
+      <ButtonGroup>
+        <Button variant="secondary" onPress={close}>Cancel</Button>
+        <Button variant="cta" onPress={close}>Register</Button>
+      </ButtonGroup>
+    </Dialog>
+  )}
+</DialogTrigger>
+```
+
+The example below illustrates how a Dialog with a hero image could be rendered via the `hero` slot:
+
+```tsx example
+<DialogTrigger>
+  <ActionButton>Upload</ActionButton>
+  {(close) => (
+    <Dialog>
+      <Image slot="hero" src="https://i.imgur.com/Z7AzH2c.png" objectFit="cover" />
+      <Heading>Upload file</Heading>
+      <Divider />
+      <Content>Are you sure you want to upload this file?</Content>
+      <ButtonGroup>
+        <Button variant="secondary" onPress={close}>Cancel</Button>
+        <Button variant="cta" onPress={close} autoFocus>Confirm</Button>
+      </ButtonGroup>
+    </Dialog>
+  )}
+</DialogTrigger>
+```
+
+### Accessibility
+
+The tab order within the Dialog follows the order of the children provided to the Dialog so keep this in mind
+when creating your Dialog. You are also responsible for determining what component, if any, should be automatically
+focused on Dialog render.
+
```

Comment:
Maybe -> "Keep in mind when creating your Dialog that the tab order within the Dialog follows the order of the children provided."